JT/14_version_control/Version_Control_README.adoc

```diff
@@ -0,0 +1,421 @@
+== 3. Resolving Conflicts ==
+
+Git conflicts are inevitable.
+
+How to solve conflicts? +
+*1.* Identify the problem +
+*2.* Choose the right resolution for each conflict (discuss with the author of the conflicting code) +
+*3.* Merge the resolved conflicts +
+
+There are a few steps that could reduce the steps needed to resolve merge conflicts in Git. +
+*Step 1:* The easiest way to resolve a conflicted file is to open it and make any necessary changes. +
+*Step 2:* After editing the file, we can use the git add command to stage the new merged content. +
+*Step 3:* The final step is to create a new commit with the help of the git commit command. +
+*Step 4:* Git will create a new merge commit to finalize the merge. +
+
+Git commands that may play a significant role in resolving conflicts. +
+*_Git Commands to Resolve Conflicts_* +
+*1. git log --merge*
+The git log --merge command helps to produce the list of commits that are causing the conflict. +
+*2. git diff*
+The git diff command helps to identify the differences between the states repositories or files. +
+*3. git checkout*
+The git checkout command is used to undo the changes made to the file, or for changing branches. +
+*4. git reset --mixed*
+The git reset --mixed command is used to undo changes to the working directory and staging area. +
+*5. git merge --abort*
+The git merge --abort command helps in exiting the merge process and returning back to the state before the merging began. +
+*6. git reset*
+The git reset command is used at the time of merge conflict to reset the conflicted files to their original state.
+
+[[gitignore]]
+== 4. .gitignore file ==
+
+Gitignore is a text file that tells Git which files or directories in a project should be ignored. This avoids cluttering your repository with unnecessary files.
+You should use a .gitignore file in each repository. It helps to ignore predefined files and directories. It may be config files, user settings, and other unwanted files.
+
+To avoid committing unnecessary files in a repository, follow these steps:
+
+1. Create a .gitignore file in the root directory of your repository. +
+2. Open the .gitignore file in a text editor. +
+3. List the filenames, directories, or file patterns that you want to exclude from version control, each on a new line. +
+4. Save the .gitignore file.
+
+Common patterns to include in .gitignore are: +
+* Directory names (for example, node_modules/, dist/, build/) +
+* File extensions (for example, *.log, *.tmp, .env) +
+* Specific files (for example, secrets.txt, config.ini) +
+
+Ensure that the .gitignore file is committed and pushed to the repository. Git will then automatically exclude the listed files and directories from being staged or committed.
+
+Regularly review and update the .gitignore file as new files or directories become unnecessary to include in the repository. This practice helps maintain a clean and focused version control history.
+
+[[best]]
+== 5. Best Practices ==
```

Comment:
I Would add:

1.  1 commit per PR
2. PR should contain all changes description
3. Pr header should be short but meaningful

Reply:
done